Inductor-spyre generated reduction SDSC doesn't match sendnn

### What happened?

While looking into the sdsc for reductions, I found that the output tensor `coordinates_` of the reduction dimension appears to differ in inductor-spyre and sendnn. Unit tests are passing, so this may not have an effect on correctness. Need to look into details to know whether it's a real bug.

Sendnn (it makes more sense that the reduction dimension is size one in the output)

<img width="497" height="457" alt="Image" src="https://github.com/user-attachments/assets/d46774bb-b909-461a-9d78-0c491ef1e359" />

inductor-spyre

<img width="541" height="455" alt="Image" src="https://github.com/user-attachments/assets/c5039154-cf4c-4f5c-8a66-24a288252cd4" />

### What did you expect to happen?

They should match, or we need to record the explanation why they don't match.

### How can we reproduce it?

Use this script and dump the sdsc
```
import torch
from einops import rearrange

torch.manual_seed(0xAFFE)
B, M, N = 3, 1024, 4096
DEVICE = torch.device("spyre")
x = torch.randn(B, M, N, dtype=torch.float16) * 0.01

DIM=1

def fn(x):
    return torch.sum(x, dim=DIM)

compiled_fn = torch.compile(fn)
z = compiled_fn(x.to(DEVICE)).cpu()

print(f"{z=}")
print(f"{z.shape=}")

fn_sendnn = torch.compile(fn, backend="sendnn")
z_sendnn = fn_sendnn(x)
print(f"{z_sendnn=}")

torch.testing.assert_close(z, z_sendnn)
```
Note that the outputs don't perfectly match.

```
z=tensor([[-0.3857, -0.0994,  0.2007,  ...,  0.0541, -0.1902, -0.1360],
        [-0.9033,  0.4380,  0.2861,  ..., -0.0185, -0.1685,  0.5391],
        [ 0.0613, -0.1045,  0.3906,  ...,  0.1492,  0.2283, -0.4243]],
       dtype=torch.float16)
z.shape=torch.Size([3, 4096])
z_sendnn=tensor([[-0.3774, -0.0962,  0.1968,  ...,  0.0500, -0.1904, -0.1333],
        [-0.9023,  0.4312,  0.2847,  ..., -0.0182, -0.1663,  0.5371],
        [ 0.0664, -0.1068,  0.3931,  ...,  0.1482,  0.2253, -0.4160]],
       dtype=torch.float16)

AssertionError: Tensor-likes are not close!

Mismatched elements: 11822 / 12288 (96.2%)
Greatest absolute difference: 0.0244140625 at index (2, 2519) (up to 1e-05 allowed)
Greatest relative difference: 35.5 at index (1, 3025) (up to 0.001 allowed)
```

### Any environmental details we need to know?

No

### Anything else we need to know?

_No response_

### Relevant log output

```Shell

```

--- a/torch_spyre/_inductor/codegen/compute_ops.py
+++ b/torch_spyre/_inductor/codegen/compute_ops.py
@@ -521,7 +521,7 @@ def generate_matmul(pointers, *, op, dimensions, inputs, outputs, **kwargs):
                                 "ldsIdx_": idx,
                                 "component_": "hbm",
                                 "layoutDimOrder_": layout_dim_order,
-                                "maxDimSizes_": [-1] * len(dim_labels),
+                                "maxDimSizes_": [-1] * len(layout_dim_order),
                                 "startAddressCoreCorelet_": {
                                     "dim_prop_func": [
                                         {"Map": {}},
@@ -749,7 +749,7 @@ def generate_bmm(pointers, *, op, dimensions, inputs, outputs, **kwargs):
                                 "ldsIdx_": idx,
                                 "component_": "hbm",
                                 "layoutDimOrder_": layout_dim_order,
-                                "maxDimSizes_": [-1] * len(dim_labels),
+                                "maxDimSizes_": [-1] * len(layout_dim_order),
                                 "startAddressCoreCorelet_": {
                                     "dim_prop_func": [
                                         {"Map": {}},

